jbang/tnb.java

```diff
@@ -0,0 +1,677 @@
+import software.tnb.common.config.TestConfiguration;
+import software.tnb.common.openshift.OpenshiftClient;
+import software.tnb.common.service.Service;
+import software.tnb.common.service.ServiceFactory;
+import software.tnb.common.utils.MapUtils;
+
+import org.junit.platform.commons.function.Try;
+import org.junit.platform.commons.util.ReflectionUtils;
+
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.jline.builtins.ConfigurationPath;
+import org.jline.builtins.Options;
+import org.jline.builtins.SyntaxHighlighter;
+import org.jline.console.CmdDesc;
+import org.jline.console.CommandInput;
+import org.jline.console.CommandMethods;
+import org.jline.console.CommandRegistry;
+import org.jline.console.ConsoleEngine;
+import org.jline.console.Printer;
+import org.jline.console.impl.Builtins;
+import org.jline.console.impl.ConsoleEngineImpl;
+import org.jline.console.impl.DefaultPrinter;
+import org.jline.console.impl.JlineCommandRegistry;
+import org.jline.console.impl.SystemHighlighter;
+import org.jline.console.impl.SystemRegistryImpl;
+import org.jline.keymap.KeyMap;
+import org.jline.reader.Binding;
+import org.jline.reader.Completer;
+import org.jline.reader.EndOfFileException;
+import org.jline.reader.LineReader;
+import org.jline.reader.LineReaderBuilder;
+import org.jline.reader.Parser;
+import org.jline.reader.Reference;
+import org.jline.reader.UserInterruptException;
+import org.jline.reader.impl.DefaultParser;
+import org.jline.reader.impl.completer.StringsCompleter;
+import org.jline.script.GroovyCommand;
+import org.jline.script.GroovyEngine;
+import org.jline.terminal.Terminal;
+import org.jline.terminal.TerminalBuilder;
+import org.jline.utils.InfoCmp;
+import org.jline.utils.OSUtils;
+import org.jline.widget.TailTipWidgets;
+import org.jline.widget.Widgets;
+import org.opentest4j.AssertionFailedError;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import groovy.lang.Closure;
+
+//CHECKSTYLE:OFF
+public class tnb {
+    //CHECKSTYLE:ON
+
+    static final List<Service<?, ?, ?>> services = new ArrayList<>();
```

Comment:
this can be left out for later, but:


should this rather be a map of `<name, service>`? when calling `deploy http` twice, it is deployed twice, however I only get one variable `localhttp`:

```
tnb> http
Service localhttp ready! You can use `localhttp` to call its methods.
tnb> http
Service localhttp ready! You can use `localhttp` to call its methods.
```